### PR TITLE
Separate Streams documentation and setup docs with easy to set variables

### PR DESCRIPTION
**Pull request by derrickdoo**
- Seperate Streams documentation out to a standalone page. 
- Setup templates to use handlebars.js
- Create template variables to swap in frequently updated values like version number from a single file templateData.js

### CODE REVIEW

File: docs/api.html [hunk 1]
```diff
@@ -15,80 +15,84 @@
...
+	Examples showing how to use the producer are given in the
+	<a href="/{{version}}/javadoc/index.html?org/apache/kafka/clients/producer/KafkaProducer.html" title="Kafka 0.10.0 Javadoc">javadocs</a>.
+	<p>
+	To use the producer, you can use the following maven dependency:
+
+	<pre>
+		&lt;dependency&gt;
+			&lt;groupId&gt;org.apache.kafka&lt;/groupId&gt;
+			&lt;artifactId&gt;kafka-clients&lt;/artifactId&gt;
+			&lt;version&gt;0.10.0.0&lt;/version&gt;
```
Comment (guozhangwang) [inline docs/api.html:42]: This is not introduced in this PR but a leftover from previous updates: 0.10.0.0 -> 0.10.1.0. Ditto elsewhere.

**Comment by guozhangwang** [comment]
@derrickdoo A minor comment, otherwise LGTM.

**Comment by guozhangwang** [comment]
@derrickdoo Could you rebase?

**Comment by derrickdoo** [comment]
@guozhangwang just rebased

File: docs/api.html (hunk 1 above)
Comment (guozhangwang) [reply docs/api.html:42]: actually could we just use the `{{version}} here`?

Comment (derrickdoo) [reply docs/api.html:42]: The current {{version}} variable follows a different pattern, but this is a great example of another variable we could introduce to our new template data file and stop making lots of find and replaces for each release. Maybe we can call it {{versionFull}}

Comment (guozhangwang) [reply docs/api.html:42]: Got it, we can work on that with a separate PR then.

**Comment by guozhangwang** [comment]
Merged to trunk.

**Pull request closed without merging**